src/shared.js

```diff
@@ -1,8 +1,21 @@
-function guid() {
-  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+function generateRandom(template) {
```

Comment:
There's a rebase necessary for this file, can you update your branch and fix the conflict?